packages/dm-core-plugins/src/form/fields/ArrayField.tsx

```diff
@@ -74,87 +68,41 @@ export default function Fields(props: any) {
   }
 
   if (onOpen && !isPrimitiveType(type)) {
-    const rows: Array<any> = []
-    const columns =
-      uiAttribute && uiAttribute.columns
-        ? [...uiAttribute.columns, 'actions']
-        : ['name', 'actions']
-    fields.map((item: any, index: number) => {
-      const row: any = {}
-      columns.forEach((column: any) => (row[column] = item[column]))
-
-      row['actions'] = (
-        <div>
-          <OpenObject
-            type={type}
-            namePath={`${namePath}.${index}`}
-            contained={true}
-            dataSourceId={dataSourceId}
-            documentId={documentId}
-            entity={item}
-          />
-          <Button
-            variant="outlined"
-            type="button"
-            onClick={() => remove(index)}
-          >
-            Remove
-          </Button>
-        </div>
-      )
-
-      rows.push(row)
-    })
-
     return (
       <Wrapper>
         <Typography bold={true}>{displayLabel}</Typography>
-        <FixedContainer>
-          <DynamicTable columns={columns} rows={rows} />
-        </FixedContainer>
-        <Button
-          variant="outlined"
-          data-testid={`add-${namePath}`}
-          onClick={() => {
-            handleAddObject()
-          }}
-        >
-          Add
-        </Button>
+        <OpenObjectButton namePath={`${namePath}`} />
```

Comment:
props to this component is of type any, but I would assume the type for namePath is a string (?)

Then it would be equivalent to write
```
<OpenObjectButton namePath={namePath} />
```